dom.bs

```diff
@@ -4185,8 +4185,8 @@ elements. SVG ought to do the same for its <{script}> elements, but does not cal
 at the moment.
 
 <p>To <dfn export id=concept-node-clone lt="clone a node" local-lt="clone">clone</dfn> a
-<var>node</var>, with an optional <var>document</var> and <i>clone children flag</i>, run these
-steps:
+<var>node</var>, with an optional <var>document</var>, <i>clone children flag</i>, and
+<i>clone shadows flag</i>, run these steps:
```

Comment:
I don't think we should allow cloning of shadow trees like this.
Instead, we should keep nested declarative shadow DOM as templates until the algorithm runs to turn it into a shadow tree. This algorithm can then recursively process inner declarative shadow DOMs.

Reply:
Ahh I see. I suppose that would work, but then you'd have a special case in the **parser** rather than just here in the cloning code, right? I'd always prefer to avoid adding more special cases to the parser if possible.

Reply:
Presumably this will have to change anyway as part of enabling streaming, right? I would expect all logic to end up in the parser, but I might be missing something.

Reply:
So I'd much prefer adding a new method like `createInstance` on `template` element to process declarative shadow DOM over declarative shadow DOM being a thing inside `template` content, and adding a special case to cloning steps.

`importNode` has a bad default of not deep cloning anyway and many people end up sub-optimally calling `template.content.cloneNode(true)` instead of `importNode`. Adding an explicit instantiating method on `template` would solve those issues.

Reply:
I think there's already confusion about the difference between `cloneNode` and `importNode`. E.g. [this SO post](https://stackoverflow.com/questions/39372886/document-importnode-vs-node-clonenode-real-example), or even [this post](https://github.com/GoogleChromeLabs/howto-components/issues/133) where strong experts aren't sure of the difference. In my own testing, I don't find a performance difference.

Having said all of that, the solution doesn't seem to be adding a third way to clone templates...